client/state/extensions-middleware/index.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * External dependencies
+ */
+import { has, invoke, mapValues } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { mergeHandlers } from '../data-layer/utils'; // FIXME: Move to this directory?
+import { middleware as wpcomApiMiddleware } from '../data-layer/wpcom-api-middleware';
+
+const MIDDLEWARES = {
+	wpcomApi: wpcomApiMiddleware
```

Comment:
the `wpcomApiMiddleware` is actually specific to the data-layer which _should_ work differently than generic middleware since we expect a chain of data layer middleware where only the first one in the chain handles a request, skipping the rest. we should consider this before bundling it in with other middleware which all need to run over the same actions.